feat: define paging for collections and items endpoint (#221)
radiantearth/stac-api-spec: ogcapi-features/README.md

Implementing OAFeat enables a wider range of clients to access the API's STAC Item objects, as it is a more widely implemented
protocol than STAC. 

## Item Pagination

OAFeat supports paging through hypermedia links for the Items resource 
(`/collections/{collectionId}/items`). Since only GET requests are allowed for this endpoint, 
it is possible
to provide a Link with `rel` type `next` and the href of the full URL of the next page of results.
This link href must contain any URL parameters that are necessary 

> for the implementation to understand how to provide the next page of results, eg: `page`, `next`,

Reviewer: ```suggestion
for the implementation to understand how to provide the next page of results, e.g. a query parameters `page`, `next`,
```

> `token`, etc. For example, the links array will have an object like:

Reviewer: ```suggestion
`token`, etc. For example, the links array will have an object like:
```
```suggestion
or `token` and any additional filter parameters if given and required. For example, the links array will have an object like:
```

```json
    "links": [
        {
            "rel": "next",
            "href": "http://api.cool-sat.com/collections/my_collection/items?page=2"
        }
    ]
```

The href may contain any arbitrary URL parameter, which is implementation-specific:

- `http://api.cool-sat.com/collections/my_collection/items?page=2`
- `http://api.cool-sat.com/collections/my_collection/items?next=8a35eba9c`
- `http://api.cool-sat.com/collections/my_collection/items?token=f32890a0bdb09ac3`

> Additionally, STAC has extended the Link object to support additional fields that provide header values

Reviewer: ```suggestion
Additionally, STAC has extended the Link object to support additional fields that provide header or other request related values
```

> to the client should they be needed for a subsequent request for the next page of results. The use

Reviewer: This sentence is confusing and potentially need a rewrite. Also, STAC did not only add header related fields, but also other request related fields such as the body.

Author: rewritten

of header values for pagination with GET requests is uncommon, so if your implementation does not use them you can
omit this attribute in the Link. These

> fields are described in detail in the [Item Search](../item-search/README.md#pagination) spec.  To avoid returning

Reviewer: ```suggestion
fields are described in detail in the [Item Search](../item-search/README.md#pagination) spec. To avoid returning
```

> the entire original request in the response, the  `merge` 

Reviewer: ```suggestion
the entire original request in the response, the `merge` 
```

property can be specified. This indicates that the client must send the same request headers that were sent in the original 
request, but with the specified headers values merged in. This allows servers to indicate what needs to change 
to get to the next page without mirroring the entire request back to the client.

Implementations may also add link relations `first` and `last`, though these are not required and may

> be infeasible to implement in some data stores.

Reviewer: I understand that `last` may be hard to implement, but `first` should be pretty easy. You just need to add the "initial" link again, without any pagination parameters.

Author: good point, i'll rewrite that.

Author: rewritten.

mechanism) through hypermedia links for the Collections resource 
(`/collections`). This mechanism aligns with pagination of collections in the 
OGC API - Common - Part 2: Geospatial Data specification. With this, a link with 
relation `next` is included in the `links` array,
and this is used to navigate to the next page of Collection objects. The specific query
parameter used for paging is implementation specific and not defined by STAC API. For example, 
an implementation may take a parameter (e.g., `page`) indicating the numeric page of results, a
base64-encoded value indicating the last result returned for the current page (e.g., `search_after` as
in Elasticsearch), or a cursor token representing backend state.  

In our simple example of numerical pages, the response for `page=3` would have a
`links` array containing these two Links indicating the URLs for the next (page=4) and 
previous (page=2) pages:

```none
"links": [
  ...
  {
    "rel": "prev",
    "href": "http://api.cool-sat.com/collections/my_collection/items?page=2"
  },
  {
    "rel": "next",
    "href": "http://api.cool-sat.com/collections/my_collection/items?page=4"
  }
]
```

> Additionally, STAC has extended the Link object to support additional fields that provide header values

Reviewer: Same as above with regards to the additional request related fields

]
```

> In addition to supporting query parameters to the `ref` URL, STAC has extended the Link object

Reviewer: `ref` should be `href`, right?

Author: yes, fixed both places.
